sklearn/ensemble/voting_classifier.py

```diff
@@ -0,0 +1,195 @@
+
+    def fit(self, X, y):
+        """ 
+        Fits the scikit-learn estimators.
+        
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape = [n_samples, n_features]
+            Training vectors, where n_samples is the number of samples and
+            n_features is the number of features.
+            
+        y : array-like, shape = [n_samples]
+            Target values.
+      
+        Returns
+        -------
+        self : object
+
+        """
+        for clf in self.clfs:
+            clf.fit(X, y)
+        return self
+            
+    def predict(self, X):
+        """
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape = [n_samples, n_features]
+            Training vectors, where n_samples is the number of samples and
+            n_features is the number of features.
+        
+        Returns
+        ----------
+        maj : array-like, shape = [n_class_labels]
+            Predicted class labels by majority rule.
+        
+        """
+        if self.voting == 'soft':
+            avg = self.predict_proba(X)
+            maj = np.apply_along_axis(lambda x: max(enumerate(x), key=operator.itemgetter(1))[0], axis=1, arr=avg)
+        
+        else: # 'hard' voting
+            self.classes_ = self._get_classes(X)
+            maj = np.asarray([np.argmax(np.bincount(self.classes_[:,c])) for c in range(self.classes_.shape[1])]) 
+        
+        return maj
+            
+    
+    def predict_proba(self, X):
+        
+        """
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape = [n_samples, n_features]
+            Training vectors, where n_samples is the number of samples and
+            n_features is the number of features.
+        
+        Returns
+        ----------
+        avg : array-like, shape = [n_samples, n_probabilities]
+            Weighted average probability for each class per sample.
+        
+        """
+        self.probas_ = self._get_probas(X)
+        avg = np.average(self.probas_, axis=0, weights=self.weights)
+        
+        return avg
+
+  
+    def transform(self, X):
+        """         
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape = [n_samples, n_features]
+            Training vectors, where n_samples is the number of samples and
+            n_features is the number of features.
+      
+        Returns
+        -------
+        If not `weights=None`:
+          array-like = [n_classifier_results, n_class_proba, n_class]
+            Class probabilties calculated by each classifier.
+        
+        Else:
+          array-like = [n_classifier_results, n_class_label]
+            Class labels predicted by each classifier.
+        
+        """
+        if self.weights:
+            return self._get_probas(X)
+        else:
+            return self._get_classes(X)  
+    
+    def _get_classes(self, X):
```

Comment:
this should be ``get_prediction`` or ``get_predict`` or just ``_predict``. The ``classes_`` attribute is ``np.unique(y)``